src/ts/contentScripts/live-preview/index.tsx

```diff
@@ -0,0 +1,169 @@
+import {Feature, Settings} from '../../utils/settings'
+import {Navigation} from '../../roam/navigation'
+import {browser} from 'webextension-polyfill-ts'
+import {createPopper, Instance} from '@popperjs/core'
+
+export const config: Feature = {
+    id: 'live_preview',
+    name: 'Live Preview',
+    defaultIsActive: false,
+    warningTooltip: 'Experimental feature; Large databases might see performance issues.',
+    settings: [],
+}
+
+Settings.isActive('live_preview').then(active => {
+    if (active) {
+        enableLivePreview()
+    }
+})
+
+browser.runtime.onMessage.addListener(async message => {
+    if (message?.featureId === 'live_preview' && message?.type === 'toggle') {
+        if (message.value) {
+            enableLivePreview()
+        } else {
+            removeLivePreview()
+        }
+    }
+})
+
+const createPreviewIframe = () => {
+    const iframe = document.createElement('iframe')
+
+    const url = Navigation.getPageUrl('search')
+    const isAdded = (pageUrl: string) => !!document.querySelector(`[src="${pageUrl}"]`)
+    if (isAdded(url)) {
+        return
+    }
+    iframe.src = url
+    iframe.style.position = 'absolute'
+    iframe.style.left = '0'
+    iframe.style.top = '0'
+    iframe.style.opacity = '0'
+    iframe.style.pointerEvents = 'none'
+
+    iframe.style.height = '0'
+    iframe.style.width = '0'
+    iframe.style.border = '0'
+    iframe.style.boxShadow = '0 0 4px 5px rgba(0, 0, 0, 0.2)'
+    iframe.style.borderRadius = '4px'
+    iframe.id = 'roam-toolkit-preview-iframe'
+
+    const styleNode = document.createElement('style')
+    styleNode.innerHTML = `
+        .roam-topbar {
+            display: none !important;
+        }
+        .roam-body-main {
+            top: 0px !important;
+        }
+        #buffer {
+            display: none !important;
+        }
+        iframe {
+            display: none !important;
+        }
+    `
+    iframe.onload = (event: Event) => {
+        ;(event.target as HTMLIFrameElement).contentDocument?.body.appendChild(styleNode)
+    }
+    document.body.appendChild(iframe)
+    const htmlElement = document.querySelector('html')
+    if (htmlElement) {
+        // to reset scroll after adding iframe
```

Comment:
why do we need this?

Reply:
some issue where after adding the iframe, the loader moved up.

Roam can fix it by putting a overflow:hidden when loading -- but until they do that, we can use this hack. Causes no harm :)

Reply:
not quite sure what you mean by "loader moved up"?  
Can you extract this into a separate function and add a more detailed explanation of "why" in the comments?

Reply:
this can happen: @Stvad  
![image](https://user-images.githubusercontent.com/2976363/82117209-5970c600-978c-11ea-95d4-be9a622c0354.png)

adding a comment too